### PR TITLE
CHE-521: Perform git init command without initial commit

### PR DESCRIPTION
**Pull request by vinokurig**
@skabashnyuk @vparfonov please review

### CODE REVIEW

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java [hunk 1]
```diff
@@ -361,16 +364,27 @@ protected void onSuccess(Branch result) {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java:335]: I think that amount of nesting levels should be reduced there.
6-7 nested blocks isn't good for reading code.

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java [hunk 2]
```diff
@@ -361,16 +364,27 @@ protected void onSuccess(Branch result) {
...
-                                                                             ? exception.getMessage()
-                                                                             : constant.branchCreateFailed();
-                                                 final GitOutputConsole console = gitOutputConsoleFactory.create(BRANCH_CREATE_COMMAND_NAME);
-                                                 console.printError(errorMessage);
-                                                 consolesPanelPresenter.addCommandOutput(appContext.getDevMachineId(), console);
-                                                 notificationManager.notify(constant.branchCreateFailed(), FAIL, true, project.getRootProject());
+                                                 if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                                     dialogFactory.createMessageDialog(constant.branchCreateNew(),
+                                                                                       constant.initCommitWasNotPerformed(),
+                                                                                       new ConfirmCallback() {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java]: No need to create an empty **ConfirmCallback** that do nothing.
You can pass null there. **ConfirmCallback** is nullable.

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java [hunk 3]
```diff
@@ -294,8 +295,18 @@ protected void onSuccess(final LogResponse result) {
 
                              @Override
                              protected void onFailure(final Throwable exception) {
-                                 Log.warn(CommitPresenter.class, "Git log failed", exception);
-                                 CommitPresenter.this.view.setMessage("");
+                                 if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                     dialogFactory.createMessageDialog(constant.branchCreateNew(), constant.initCommitWasNotPerformed(),
+                                                                       new ConfirmCallback() {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java]: the same as https://github.com/eclipse/che/pull/473#discussion_r53746239

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java [hunk 4]
```diff
@@ -149,7 +151,18 @@ protected void onSuccess(LogResponse result) {
 
                            @Override
                            protected void onFailure(Throwable exception) {
-                               notificationManager.notify(locale.logFailed(), FAIL, false);
+                               if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                   dialogFactory.createMessageDialog(locale.compareWithRevisionTitle(), locale.initCommitWasNotPerformed(),
+                                                                     new ConfirmCallback() {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java]: the same as https://github.com/eclipse/che/pull/473#discussion_r53746239

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java [hunk 5]
```diff
@@ -165,12 +173,25 @@ protected void onSuccess(LogResponse result) {
...
                         protected void onFailure(Throwable exception) {
-                            nothingToDisplay(null);
-                            String errorMessage = exception.getMessage() != null ? exception.getMessage() : constant.logFailed();
-                            GitOutputConsole console = gitOutputConsoleFactory.create(LOG_COMMAND_NAME);
-                            console.printError(errorMessage);
-                            consolesPanelPresenter.addCommandOutput(appContext.getDevMachineId(), console);
-                            notificationManager.notify(constant.logFailed(), FAIL, true, project);
+                            if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                dialogFactory.createMessageDialog(constant.historyTitle(), constant.initCommitWasNotPerformed(),
+                                                                  new ConfirmCallback() {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java]: the same as https://github.com/eclipse/che/pull/473#discussion_r53746239

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java [hunk 6]
```diff
@@ -103,11 +111,22 @@ protected void onSuccess(LogResponse result) {
...
                         @Override
                         protected void onFailure(Throwable exception) {
-                            String errorMessage = (exception.getMessage() != null) ? exception.getMessage() : constant.logFailed();
-                            GitOutputConsole console = gitOutputConsoleFactory.create(LOG_COMMAND_NAME);
-                            console.printError(errorMessage);
-                            consolesPanelPresenter.addCommandOutput(appContext.getDevMachineId(), console);
-                            notificationManager.notify(constant.logFailed(), FAIL, true, appContext.getCurrentProject().getRootProject());
+                            if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                dialogFactory.createMessageDialog(constant.resetCommitViewTitle(), constant.initCommitWasNotPerformed(),
+                                                                  new ConfirmCallback() {
```
Comment (azatsarynnyy) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java]: the same as https://github.com/eclipse/che/pull/473#discussion_r53746239

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java (hunk 1 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java:335]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java (hunk 2 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java (hunk 3 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java (hunk 4 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java (hunk 5 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java (hunk 6 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java]: done

**Comment by skabashnyuk** [comment]
ok. @sleshchenko ?

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java [hunk 7]
```diff
@@ -294,8 +295,14 @@ protected void onSuccess(final LogResponse result) {
...
                              @Override
                              protected void onFailure(final Throwable exception) {
-                                 Log.warn(CommitPresenter.class, "Git log failed", exception);
-                                 CommitPresenter.this.view.setMessage("");
+                                 if (getErrorCode(exception) == ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED) {
+                                     dialogFactory.createMessageDialog(constant.commitTitle(),
+                                                                       constant.initCommitWasNotPerformed(),
+                                                                       null).show();
+                                 } else {
+                                     Log.warn(CommitPresenter.class, "Git log failed", exception);
```
Comment (sleshchenko) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java:305]: Think about showing message for user. Something like

```
notificationManager.notify(locale.logFailed(), FAIL, false);
```

**Comment by sleshchenko** [comment]
other LGTM

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java (hunk 7 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java:305]: done

**Comment by vparfonov** [comment]
ok

**Pull request merged**